lib/rbmysql/protocol.rb

```diff
@@ -502,7 +506,7 @@ def read
           f, errno, message = data.unpack("Cva*")    # Version 4.0 Error
           @sqlstate = ""
         end
-        message.force_encoding(@charset.encoding)
+        message.force_encoding(@charset.encoding) if @charset
```

Comment:
Fixes the following error:

```
[-] 127.0.0.1:3306        - 127.0.0.1:3306 Error: undefined method `encoding' for nil:NilClass
```

Which can be caused by immediately receiving an error message from the server without negotiating a charset from the server, from wireshark:

```
MySQL Protocol
    Packet Length: 104
    Packet Number: 0
    Error Code: 1129
    Error message: Host '172.17.0.1' is blocked because of many connection errors; unblock with 'mysqladmin flush-hosts'
```

After handling the nil scenario the user will now get the actual server error:

```
msf6 auxiliary(scanner/mysql/mysql_authbypass_hashdump) > rerun rhost=127.0.0.1
[*] Reloading module...

[-] 127.0.0.1:3306        - 127.0.0.1:3306 Error: Host '172.17.0.1' is blocked because of many connection errors; unblock with 'mysqladmin flush-hosts'
[*] 127.0.0.1:3306        - Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
```

This fix also allows the `RbMysql::HostNotPrivileged` exception to trigger as expected, previously this would give the same charset exception - now it gives the real error to the user:

```
[-] 10.35.1.23:3306       - 10.35.1.23:3306 Unable to login from this host due to policy (may still be vulnerable)
```